lib/march_hare/session.rb

```diff
@@ -318,7 +322,8 @@ def disable_automatic_recovery
     # Begins automatic connection recovery (typically only used internally
     # to recover from network failures)
     def automatically_recover
-      @logger.debug("session: begin automatic connection recovery")
+      raise ConnectionClosedException if @was_explicitly_closed
```

Comment:
I tried to make this new behaviors more obvious by raising a exception if `automatically_recover` is called after an explicit close.